### PR TITLE
actually applies the artifical norm this time, bases the A.N. on redu…

### PR DESCRIPTION
**Pull request by walshmm**
…ction data as per Malcolm, makes A.N. for EACH group, smoothing field allows for appropriate value >1

## Description of work

This reorgs and applies things so that Artificial Normalization is effective during reduction.

## To test

Refer to #487 for testing params but refer to this ticket for acceptance criteria.

## Link to EWM item

[EWM#8026](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=8026)

### Verification

- [x] the author has read the EWM story and acceptance critera
- [x] the reviewer has read the EWM story and acceptance criteria
- [ ] the reviewer certifies the acceptance criteria below reflect the criteria in EWM

### Acceptance Criteria

This list is for ease of reference, and does not replace reading the EWM story as part of the review.  Verify this list matches the EWM story before reviewing.

- [x]  Artificial Normalization is actually applied to reduction results. (check this with an extreme artificial normalization.)
- [x]  Smoothing Parameter is a text field to allow freedom to test appropriate parameters for a future slider.
- [x]  Artificial Normalization should be based on the input run for reduction, and one should be made for each group.
- [x]  Due to limitations of the ui, the preview Artificial Normalization will use a Column grouping when deciding the parameters.
- [x]  Lacking a calibration should not prevent a user from proceeding with and Artificial Normalization

### CODE REVIEW

File: src/snapred/backend/dao/request/ReductionRequest.py
Comment (darshdinger) [inline src/snapred/backend/dao/request/ReductionRequest.py:1]: This makes a lot more sense.

**Comment by darshdinger** [comment]
# Testing Overview:
--------------------------
1. Attempt Reduction without state initialized leads to this error:
image

2. Proceed with fresh state:
image
Then this error once the processing is complete:
image
The workflow succeeds regardless:
image

3. Trial with diffcal present:
I get this error and no artificial normalization pop up for no normalization present:
image

4. Trial with both normcal and diffcal present:
Reduction seems to proceed as normal:
image


Based on my tests, there are some present issues with the error pop ups for notifying the user.

**Comment by walshmm** [comment]
1. I need to reword the error since it trys to punt you to the state init pop up but you dont have write perms.  It shouldnt be referencing the reduction save path.
2. I think I just need to silence that **warning**?  I'm not sure its actionable from the user or relevant.


These _warnings_ shouldnt be blocking.
Please proceed in reviewing the data and gui functionality as outlined in the AC while I address the inaccurate error message and noisy warning pop up.

**Comment by darshdinger** [comment]
# Testing AC:
------------------

1. Run Reduction without Normalization:
image

2. Testing Smoothing Parameter:
Original - 
image
Trying a different smoothing parameter -
image

The artificial normalization changed a bit in terms of the smoothness but in all cases attempted, it doesn't match up to the diffcal data. Column grouping is present as per AC. Text field is usable and functional as per AC.

4. Final Result:
image
Seems like it has been applied to each grouping schema.

5. As tested before, lacking a calibration does not prevent a user from proceeding with an artificial normalization.

**Review by darshdinger — APPROVED** [review]
Through retesting of PR after additional changes results in the expected behavior. The implementation of additional exceptions is functional.

**Pull request merged**